### PR TITLE
Adding the SC21 Lab3 content for CI/CD pipeline

### PR DESCRIPTION
**Pull request by kraman-aws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/04-cicd-pipeline/02-codecommit-repo.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
++++
+title = "a. Create a repo in CodeCommit"
+date = 2021-09-30T10:46:30-04:00
+weight = 30
```
Comment (mhuguesaws) [inline content/04-cicd-pipeline/02-codecommit-repo.md:4]: Should be updated

File: content/04-cicd-pipeline/_index.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#### Test and automate infrastructure
+
+Combine infrastructure as code (IaC) with version control and automated, continuous integration to bring scalability and consistency to provisioning and management.
+
+{{% notice info %}}This lab requires an AWS Cloud9 IDE. If you do not have an AWS Cloud9 IDE set up, complete sections *a. Sign in to the Console* through *d. Work with the AWS CLI* in the [**Getting Started in the Cloud**](/02-aws-getting-started.html) workshop.
+{{% /notice %}}
+
+In this lab, you are introduced to [AWS Developer Tools](https://aws.amazon.com/products/developer-tools/) and how to use services like AWS CodeCommit, AWS CodeBuild and AWS CodePipeline to automate application deployment with containers, CI/CD pipelines and container orchestrators . This workshop includes the following steps:
+
+- Create a repository in AWS CodeCommit
```
Comment (perifaws) [inline content/04-cicd-pipeline/_index.md:38]: Numbered list

File: content/04-cicd-pipeline/_index.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+title: "Create a CI/CD pipeline"
+date: 2021-09-29
+weight: 50
+pre: "<b>III ⁃ </b>"
+tags: ["CI/CD", "AWS Developer Tools", "DevOps"]
+---
+
+![devops_logo](/images/cicd/devops-logo.png)
+
+[AWS Developer Tools](https://aws.amazon.com/products/developer-tools/) provides a list of services to host code, build, test, and deploy your applications quickly and effectively.  AWS services offered as part of the AWS Developer Tools suite helps remove the undifferentiated heavy lifting associated with DevOps adaptation and software development. You can build a continuous integration and delivery capability without managing servers or build nodes, and leverage Infrastructure as Code to provision and manage your cloud resources in a consistent and repeatable manner.
```
Comment (perifaws) [inline content/04-cicd-pipeline/_index.md:11]: Where is the HPC context here?

File: content/04-cicd-pipeline/02-codecommit-repo.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+{{% notice info %}}This lab requires an AWS Cloud9 IDE. If you do not have an AWS Cloud9 IDE set up, complete sections *a. Sign in to the Console* through *d. Work with the AWS CLI* in the [**Getting Started in the Cloud**](/02-aws-getting-started.html) workshop.
+{{% /notice %}}
+
+1. In the AWS Management Console search bar, type and select **Cloud9**.
+
+2. Choose **open IDE** for the Cloud9 instance set up previously. It may take a few moments for the IDE to open. AWS Cloud9 stops and restarts the instance so that you do not pay compute charges when no longer using the Cloud9 IDE.
+
+3. Next, we'll use the AWS CLI to create a Git repository in [AWS CodeCommit](https://aws.amazon.com/codecommit/) and clone the empty repo in your Cloud9 environment.
+
+1. First verify AWS CLI is installed:
```
Comment (perifaws) [inline content/04-cicd-pipeline/02-codecommit-repo.md:17]: First time you talk about the CLI (AWS Command Line Interface (CLI))

File: content/04-cicd-pipeline/02-codecommit-repo.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+```bash
+aws --version
+```
+
+2. Next create your CodeCommit Repo:
+
+AWS CodeCommit is a secure, highly scalable, managed source control service that hosts private Git repositories.
```
Comment (perifaws) [inline content/04-cicd-pipeline/02-codecommit-repo.md:25]: Or provide the link to the console page

File: content/04-cicd-pipeline/02-codecommit-repo.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+```bash
+aws codecommit create-repository --repository-name MyDemoRepo --repository-description "My demonstration repository" --tags Team=SC21
+```
+
+3. Get repository URL to clone:
```
Comment (perifaws) [inline content/04-cicd-pipeline/02-codecommit-repo.md:31]: No say that you can get the URL to your repo, not for cloning yet

File: content/04-cicd-pipeline/02-codecommit-repo.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+```bash
+REPOURL=$(aws codecommit get-repository --repository-name MyDemoRepo --query repositoryMetadata.cloneUrlHttp --output text)
+echo $REPOURL
+```
+
+Verify `echo $REPOURL` outputs a repo url like: `https://git-codecommit.<region>.amazonaws.com/v1/repos/MyDemoRepo`
```
Comment (perifaws) [inline content/04-cicd-pipeline/02-codecommit-repo.md:38]: where region is X, Z Y

File: content/04-cicd-pipeline/02-codecommit-repo.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+4. Clone the repository in your Cloud9 terminal and cd into it:
```
Comment (perifaws) [inline content/04-cicd-pipeline/02-codecommit-repo.md:40]: go into it using the command `cd`

File: content/04-cicd-pipeline/02-codecommit-repo.md (hunk 7 above)
Comment (perifaws) [reply content/04-cicd-pipeline/02-codecommit-repo.md:38]: repository URL

File: content/04-cicd-pipeline/02-codecommit-repo.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+```bash
+$ git clone $REPOURL
+Cloning into 'MyDemoRepo'...
+remote: Counting objects: 4, done.
+Unpacking objects: 100% (4/4), 681 bytes | 227.00 KiB/s, done.
+$ cd MyDemoRepo/
+```
+
+5. Now let's update the default branch from **master** to **main**:
```
Comment (perifaws) [inline content/04-cicd-pipeline/02-codecommit-repo.md:50]: Use code notation for `master` and `main`. By the way, why do you do that?

File: content/04-cicd-pipeline/07-conclusion.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
++++
+title = "e. Conclusion"
```
Comment (perifaws) [inline content/04-cicd-pipeline/07-conclusion.md:2]: Should be f

File: content/04-cicd-pipeline/03-docker-buildspec.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
++++
+title = "b. Create Docker and buildspec files"
+date = 2021-09-30T10:46:30-04:00
+weight = 40
+tags = ["tutorial", "DeveloperTools", "CodeCommit"]
++++
+
+In this section, you will create a Docker container for the application and a buildspec file in the CodeCommit repository created in the previous section
+
+A [buildspec](https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html) is a collection of build commands and related settings in YAML format. This file is used by [AWS CodeBuild](https://docs.aws.amazon.com/codebuild/latest/userguide/welcome.html) to automatically create an updated version of the container upon code changes. In the next section, you will dive deeper on what is CodeBuild and how to set it up as part of a pipeline.
```
Comment (perifaws) [inline content/04-cicd-pipeline/03-docker-buildspec.md:10]: > A [buildspec](https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html) is a collection of build commands and related settings in YAML format

Provide an example, what does this file do exactly?

File: content/04-cicd-pipeline/03-docker-buildspec.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+weight = 40
+tags = ["tutorial", "DeveloperTools", "CodeCommit"]
++++
+
+In this section, you will create a Docker container for the application and a buildspec file in the CodeCommit repository created in the previous section
+
+A [buildspec](https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html) is a collection of build commands and related settings in YAML format. This file is used by [AWS CodeBuild](https://docs.aws.amazon.com/codebuild/latest/userguide/welcome.html) to automatically create an updated version of the container upon code changes. In the next section, you will dive deeper on what is CodeBuild and how to set it up as part of a pipeline. 
+
+
+1. cd to the CodeCommit repository created in the previous section:
```
Comment (perifaws) [inline content/04-cicd-pipeline/03-docker-buildspec.md:13]: is `cd` a verb?

File: content/04-cicd-pipeline/03-docker-buildspec.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+```bash
+cd MyDemoRepo
+```
+
+2. Create a Docker container for the application. We're going to use the [Amazon Linux container](https://gallery.ecr.aws/amazonlinux/amazonlinux) from [Amazon Elastic Container Registry (ECR) Public Gallery](https://docs.aws.amazon.com/AmazonECR/latest/public/public-gallery.html). Amazon Linux Docker container images contain a subset of the packages in the images for use on Amazon EC2 and as VMs in on-premises scenarios. To install additional packages, use yum.
```
Comment (perifaws) [inline content/04-cicd-pipeline/03-docker-buildspec.md:19]: `yum`?

File: content/04-cicd-pipeline/03-docker-buildspec.md [hunk 14]
```diff
@@ -0,0 +1,85 @@
++++
```
Comment (perifaws) [inline content/04-cicd-pipeline/03-docker-buildspec.md:1]: any image you could show as an example of result?

**Pull request merged**